app/models/snap_application.rb

```diff
@@ -14,6 +14,12 @@ def self.enqueue_faxes
     end
   end
 
+  def pdf
+    @pdf ||= Dhs1171Pdf.new(
```

Comment:
nice! would this memoization cause problems if we were to update the snap app and want the PDF updated as well?

Reply:
It could, but it is unlikely as every request/response cycle retrieves a snap application from the database. (This may not be as true with action cable tho?). We could also do a `dirty?` check on the applicant information and force regeneration when the application is dirty.